test/integration/framework.go

```diff
@@ -69,45 +70,56 @@ func withConfigGetFreshApiserverAndClient(
 	*restclient.Config,
 	func(),
 ) {
-	securePort := rand.Intn(31743) + 1024
-	secureAddr := fmt.Sprintf("https://localhost:%d", securePort)
 	stopCh := make(chan struct{})
 	serverFailed := make(chan struct{})
+
+	certDir, _ := ioutil.TempDir("", "service-catalog-integration")
+	secureServingOptions := genericserveroptions.NewSecureServingOptions()
+
+	var etcdOptions *server.EtcdOptions
+	if serverstorage.StorageTypeEtcd == serverConfig.storageType {
+		etcdOptions = server.NewEtcdOptions()
+		etcdOptions.StorageConfig.ServerList = serverConfig.etcdServerList
+		etcdOptions.EtcdOptions.StorageConfig.Prefix = fmt.Sprintf("%s-%08X", server.DefaultEtcdPathPrefix, rand.Int31())
```

Comment:
curiosity question, why is it called int31 to get in int32? Is it immediately obvious to everyone else?

Reply:
Not without reviewing the Int31() function comment:  "Int31 returns a non-negative pseudo-random 31-bit integer as an int32."